File: Selenium/Activities/Activity3.py
# Repeat the Activity1 & Activity2 but strictly use xpaths in your locators.

#import webdriver
from selenium import webdriver
from selenium.webdriver.common.by import By

#Start the driver
with webdriver.Firefox() as driver:
    #Navigate to the url
    driver.get("https://training-support.net/webelements/login-form")
    #Print the title of the page
    print("Page title: ", driver.title)

    #Find the username field
    username = driver.find_element(By.XPATH, "//input[@id='username']")
    #Find the password field
    password = driver.find_element(By.XPATH, "//input[@id='password']")
    
    #Enter the username and password
    username.send_keys("admin")
    password.send_keys("password")

    #FInd the login button
    login = driver.find_element(By.XPATH, "//button[text()='Submit']")
    login.click()

    #Print the login message
    message = driver.find_element(By.XPATH, "//h1[contains(@class, text-center)]")
    print("Login message: ", message.text)


    

File: TestNG/Activities/Activity3.java
/*Create a TestNG Class with the annotations
@Test
@BeforeClass
@AfterClass
In the @BeforeClass method, create the a driver instance for FirefoxDriver
Also use the get() method to open the browser with https://training-support.net/webelements/login-form/
In the @AfterClass method, use close() to close the browser once the test is done.
Find the login input fields - username and password
Enter the credentials - admin and password - and click the "Log in" button.
Read the confirmation message and make and assertion to check if the correct message has been printed.
*/

package activities;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.Test;
import org.testng.annotations.BeforeClass;

public class Activity3 {

	WebDriver driver;
	WebDriverWait wait;
	
	@BeforeClass
	public void homePageTest() {
		//Create an instance to firefox driver
		driver = new FirefoxDriver();
		
		//Initialize the wait
		wait = new WebDriverWait(driver, Duration.ofSeconds(10));
		
		//Open the browser
		driver.get("https://training-support.net/webelements/login-form");
	}
	@Test
	public void login() {
		//Find the web element of username and password
		WebElement username = driver.findElement(By.id("username"));
		WebElement password = driver.findElement(By.id("password"));
		
		//Enter the login credentials in the fields username and password. 
		username.sendKeys("admin");
		password.sendKeys("password");
		
		//Find the Submit button and click it.
		WebElement login = driver.findElement(By.xpath("//button[text()='Submit']"));
		login.click();
	}
	@Test
	public void assertionTest() {
		String loginMsg = driver.findElement(By.xpath("//h2[contains(@class, 'text-center')]")).getText();
		System.out.println("Login message: " + loginMsg);
		Assert.assertEquals(loginMsg, "Welcome Back, Admin!");
	}
	
	@AfterClass
	public void closingBrowser() {
		driver.close();
	}
}
